var/spack/repos/builtin/packages/turbovnc/package.py

```diff
@@ -0,0 +1,115 @@
+
+    variant('java',    default=False, description='Enable Java build')
+    variant('server',  default=True,  description='Enable server build')
+    variant('x11deps', default=True,  description='Depends x11 depends')
+
+    patch('x11deps.patch', when='@2.1:')
+
+    depends_on("libx11", when='+x11deps')
+    depends_on("libjpeg-turbo@1.3.1", when='@1.2.2')
+    depends_on("libjpeg-turbo@1.5.1", when='@2.1:')
+    depends_on("libjpeg-turbo@1.5.1+java", when='@2.1:+java',)
+    depends_on('jdk', when='+java')
+    depends_on('openssl', when='@2.2:')
+    depends_on('openssl@:1.0.999', when='@:2.1')
+    depends_on("pam")
+    depends_on("libxext", when='+x11deps')
+    depends_on("libxdmcp", when='+x11deps')
+    depends_on("libxau", when='+x11deps')
+    depends_on("libxdamage", when='+x11deps')
+    depends_on("libxcursor", when='+x11deps')
+    depends_on("libxi", when='+java')
+    depends_on("libxt", when='+java')
+    depends_on("libxxf86vm", when='+x11deps')
+    depends_on("libxxf86misc", when='+x11deps')
+    depends_on("xf86vidmodeproto", when='+x11deps')
+    # depends_on('libxkbfile')
+    depends_on('xkeyboard-config', when='+x11deps', type=('build', 'run'))
+    depends_on('xkbcomp', when='+x11deps', type=('build', 'run'))
+    depends_on('xauth', when='+x11deps', type=('build', 'run'))
+    # depends_on('xkbdata', when='+x11deps', type='build')
+
+    def validate(self):
+        """
+        Checks if incompatible versions of openssl were specified
+
+        :param spec: spec of the package
+        :raises RuntimeError: in case of inconsistencies
+        """
+        spec = self.spec
+        if spec.satisfies('@:2.1') and spec.satisfies('^openssl@1.1:'):
+            msg = 'turbovnc does not compile with openssl 1.1 '
+            raise RuntimeError(msg)
+
+    def cmake_args(self):
+
+        self.validate()
+        options = []
+        if '+java' in self.spec:
+            options.append('-DTVNC_BUILDJAVA:BOOL=ON')
+            options.append(
+                '-DTJPEG_JAR:PATH=' + self.spec['libjpeg-turbo'].prefix +
+                '/share/classes/turbojpeg.jar')
+            options.append(
+                '-DTJPEG_JNILIBRARY:PATH=' +
+                self.spec['libjpeg-turbo'].prefix +
+                '/lib/libturbojpeg.so')
+        else:
+            options.append('-DTVNC_BUILDJAVA:BOOL=OFF')
+            options.append('-DTVNC_BUILDNATIVE:BOOL=ON')
+        if '+server' in self.spec:
+            options.append('-DTVNC_BUILDSERVER:BOOL=ON')
+            if '~x11deps' in self.spec:
+                options.append('-DTVNC_NVCONTROL:BOOL=ON')
+        if '+x11deps' in self.spec:
+            options.append('-DXKB_BASE_DIRECTORY:PATH=' +
+                           self.spec['xkeyboard-config'].prefix +
+                           '/share/X11/xkb')
+            options.append('-DXKB_BIN_DIRECTORY:PATH=' +
+                           self.spec['xkbcomp'].prefix + '/bin')
+        if '+debug' in self.spec:
```

Comment:
There is no `debug` variant for this package. Spack's CMakePackage base class already handles `CMAKE_BUILD_TYPE` for you, so you can remove this if-block.